File: CS/MvpvmNavigation/Views/ModuleB/View2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.XtraBars.Ribbon;
using DevExpress.Mvvm;
using MvpvmNavigation.ViewModels;

namespace MvpvmNavigation.Views {
    public partial class View2 : BaseModuleControl, IRibbonModule {
        public View2()
            : base(CreateViewModel<ModuleBViewModel>) {
            InitializeComponent();
            BindCommands();
            System.Threading.Thread.Sleep(1000);
        }
        protected override void OnInitServices(IServiceContainer serviceContainer) {
            serviceContainer.RegisterService(new Services.EditFormService(Model.ModuleType.ModuleB_EditForm));
        }
        void BindCommands() {
            buttonShow.BindCommand(() => ViewModel.ShowEditForm(), ViewModel);
        }
        public ModuleBViewModel ViewModel {
            get { return GetViewModel<ModuleBViewModel>(); }
        }
        RibbonControl IRibbonModule.Ribbon {
            get { return ribbonControl1; }
        }
    }
}

File: VB/MvpvmNavigation/Views/ModuleB/View2.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.Mvvm
Imports MvpvmNavigation.ViewModels

Namespace MvpvmNavigation.Views
    Partial Public Class View2
        Inherits BaseModuleControl
        Implements IRibbonModule

        Public Sub New()
            MyBase.New(AddressOf CreateViewModel(Of ModuleBViewModel))
            InitializeComponent()
            BindCommands()
            System.Threading.Thread.Sleep(1000)
        End Sub
        Protected Overrides Sub OnInitServices(ByVal serviceContainer As IServiceContainer)
            serviceContainer.RegisterService(New Services.EditFormService(Model.ModuleType.ModuleB_EditForm))
        End Sub
        Private Sub BindCommands()
            buttonShow.BindCommand(Sub() ViewModel.ShowEditForm(), ViewModel)
        End Sub
        Public ReadOnly Overloads Property ViewModel() As ModuleBViewModel
            Get
                Return GetViewModel(Of ModuleBViewModel)()
            End Get
        End Property
        Private ReadOnly Property IRibbonModule_Ribbon() As RibbonControl Implements IRibbonModule.Ribbon
            Get
                Return ribbonControl1
            End Get
        End Property
    End Class
End Namespace
